src/super_gradients/training/models/predictions.py

```diff
@@ -0,0 +1,75 @@
+from typing import Tuple
+from abc import ABC
+from dataclasses import dataclass
+
+import numpy as np
+
+from super_gradients.common.factories.bbox_format_factory import BBoxFormatFactory
+from super_gradients.training.datasets.data_formats.bbox_formats import convert_bboxes
+
+
+@dataclass
+class Prediction(ABC):
+    pass
+
+
+@dataclass
+class DetectionPrediction(Prediction):
+
+    _bboxes: np.ndarray
+    _bbox_format: str
+
+    confidence: np.ndarray
+    labels: np.ndarray
+    image_shape: Tuple[int, int]
+
+    def __init__(self, bboxes: np.ndarray, bbox_format: str, confidence: np.ndarray, labels: np.ndarray, image_shape: Tuple[int, int]):
+        """
+        :param bboxes:      BBoxes in the format specified by bbox_format
+        :param bbox_format: BBoxes format that can be a string ("xyxy", "cxywh", ...)
+        :param confidence:  Confidence scores for each bounding box
+        :param labels:      Labels for each bounding box
+        :param image_shape: Shape of the image the prediction is made on
+        """
+        self._bboxes = bboxes
+        self._bbox_format = bbox_format
+        self.confidence = confidence
+        self.labels = labels
+        self.image_shape = image_shape
+
+    @property
+    def bboxes_xyxy(self):
+        return self._get_bbox_as("xyxy")
+
+    @bboxes_xyxy.setter
+    def bboxes_xyxy(self, bboxes: np.ndarray):
+        self._set_bbox_from(bboxes=bboxes, input_bbox_format="xyxy")
+
+    @property
+    def bboxes_cxcywh(self):
```

Comment:
A getter and setter for bboxes_cxcywh is not used anywhere and can (and should IMO) be removed.
You don't need this format for visualization and if someone needs boxes in format other than XYXY he/she can use convert_bboxes call outside the DetectionPrediction. IMO this class should not be responsible to give boxes to user in custom formats.

Reply:
I replied about this in another comment